analytic_engine/src/table/data.rs

```diff
@@ -69,12 +75,108 @@ pub enum Error {
     FindMemTable {
         source: crate::table::version::Error,
     },
+
+    #[snafu(display("Failed to persist max file id, err:{}", source))]
+    PersistFileId { source: GenericError },
 }
 
 define_result!(Error);
 
 pub type MemTableId = u64;
 
+const ALLOC_STEP: u64 = 1000;
+
+#[derive(Debug)]
+pub struct Allocator {
```

Comment:
Maybe it should be named as `IdAllocator`, and it can be put in the `common_util` package. We need some unit tests for it.